src/viper/pretty.ml

```diff
@@ -98,23 +100,24 @@ and pp_exp ppf exp =
      fprintf ppf "%s" (Mo_values.Numerics.Int.to_string i)
 
 and pp_stmt ppf stmt =
+  marks := stmt.at :: !marks;
   match stmt.it with
   | SeqnS seqn -> pp_seqn ppf seqn
   | IfS(exp1, s1, { it = ([],[]); _ }) ->
-    fprintf ppf "@[<v 2>if %a@ %a@]"
+    fprintf ppf "\017@[<v 2>if %a@ %a@]\019"
       pp_exp exp1
       pp_seqn s1
   | IfS(exp1, s1, s2) ->
-    fprintf ppf "@[<v 2>if %a@ %aelse@ %a@]"
+    fprintf ppf "\017@[<v 2>if %a@ %aelse@ %a@]\019"
       pp_exp exp1
       pp_seqn s1
       pp_seqn s2
   | VarAssignS(id, exp) ->
-    fprintf ppf "@[<v 2>%s := %a@]"
+    fprintf ppf "\017@[<v 2>%s := %a@]\019"
       id.it
       pp_exp exp
   | FieldAssignS(fldacc, exp2) ->
-    fprintf ppf "@[<v 2>%a := %a@]"
+    fprintf ppf "\017@[<v 2>%a := %a@]\019"
```

Comment:
Can we move the marking insertion into a helper?
let pp_stmt = ... push marks ... frpintf ppf "\017%a\019" pp_stmt' exp ...
let pp_stmt' = ...
(removing the markers from each case?